tsdb/index/tsi1/file_set.go

```diff
@@ -808,21 +729,28 @@ func (fs *FileSet) MeasurementSeriesKeysByExpr(name []byte, expr influxql.Expr,
 
 	// Iterate over all series and generate keys.
 	var keys [][]byte
-	for e := itr.Next(); e != nil; e = itr.Next() {
+	for e := itr.Next(); e.SeriesID != 0; e = itr.Next() {
 		// Check for unsupported field filters.
 		// Any remaining filters means there were fields (e.g., `WHERE value = 1.2`).
-		if e.Expr() != nil {
-			if v, ok := e.Expr().(*influxql.BooleanLiteral); !ok || !v.Val {
+		if e.Expr != nil {
+			if v, ok := e.Expr.(*influxql.BooleanLiteral); !ok || !v.Val {
 				return nil, errors.New("fields not supported in WHERE clause during deletion")
 			}
 		}
 
-		keys = append(keys, models.MakeKey(e.Name(), e.Tags()))
+		seriesKey := fs.sfile.SeriesKey(e.SeriesID)
+		assert(seriesKey != nil, "series key not found")
```

Comment:
Can we change this to an `error`? I'm not keen on us panicking if we have the chance to return an error.

Reply:
Maybe we could add something about breaking an invariant to the error messages if we want to clear way to determine that the error is really significant?